lib/src/resizable_container.dart

```diff
@@ -58,15 +58,20 @@ class _ResizableContainerState extends State<ResizableContainer> {
   void initState() {
     super.initState();
 
-    manager.setChildren(widget.children);
+    controller.setChildren(widget.children);
   }
 
   @override
   void didUpdateWidget(covariant ResizableContainer oldWidget) {
-    final hasChanges = !listEquals(oldWidget.children, widget.children);
+    final hasChanges = !listEquals(oldWidget.children, widget.children)
+      || oldWidget.direction != widget.direction;
 
     if (hasChanges) {
       manager.updateChildren(widget.children);
+      keys = List.generate(
```

Comment:
Not sure if it's entirely worthwhile, but we could add a constraint to only regenerate this list if the children change and not if _only_ the direction changes.

Reply:
According to the docs, it is relatively expensive, but it only warns against recreating these in build. This method shouldn't be called that often, and when it is, there's a lot more expensive work going on